inst/tests/test-sanitise-dim.r

```diff
@@ -0,0 +1,76 @@
+test_that(
+  "sanitise_dim returns NULL for zero-length inputs, with appropriate warnings",
```

Comment:
This seems like way too many unit tests for such a simple function. Could you please simplify focussing on the highest risk areas?